src/sui/SuiTools.ts

```diff
@@ -55,8 +55,23 @@ export class SuiTools implements EdgeCurrencyTools {
     privateWalletInfo: EdgeWalletInfo
   ): Promise<string> {
     const { pluginId } = this.currencyInfo
-    const keys = asSuiPrivateKeys(pluginId)(privateWalletInfo.keys)
-    return keys.mnemonic
+    const mnemonic: unknown = privateWalletInfo.keys[`${pluginId}Mnemonic`]
+    if (typeof mnemonic === 'string') return mnemonic
+
+    const hexKey: unknown = privateWalletInfo.keys[`${pluginId}Key`]
+    if (typeof hexKey === 'string') {
+      const hex = hexKey.replace(/^0x/i, '').toLowerCase()
+      const keyBytes = Buffer.from(hex, 'hex')
+      if (keyBytes.length !== 32) {
+        throw new Error('InvalidPrivateKey')
+      }
+      const data = new Uint8Array(1 + keyBytes.length)
+      data[0] = 0x00 // ed25519 scheme flag
+      data.set(keyBytes, 1)
+      return toBech32('suiprivkey', data)
```

Comment:
First, toBech32 is imported from the cosmjs library. That itself is not necessary wrong, since bech32 is an encoding, but we should instead use a general library or a sui-specific library.

Second, we should display the key as entered even if we use a different version of it internally. You could show both, but definitely show what the user entered.

Reply:
We already use bench32 functions from the cosmjs lib in other currencies,
figured we want to continue and not add extra dependencies. Regardless, removed
this bit anyway.